generators/go-v2/dynamic-snippets/src/context/DynamicSnippetsGeneratorContext.ts

```diff
@@ -72,65 +80,95 @@ export class DynamicSnippetsGeneratorContext {
-            instances.push({
-                name: parameter.name.name,
-                typeReference: parameter.typeReference,
-                value
-            });
         }
         return instances;
     }
 
-    public getRecordOrThrow(value: unknown): Record<string, unknown> {
+    public getRecord(value: unknown): Record<string, unknown> | undefined {
         if (typeof value !== "object" || Array.isArray(value)) {
-            throw new Error(`Expected object with key, value pairs but got: ${JSON.stringify(value)}`);
+            this.errors.add({
+                severity: Severity.Critical,
+                message: `Expected object with key, value pairs but got: ${JSON.stringify(value)}`
+            });
+            return undefined;
         }
         if (value == null) {
             return {};
         }
         return value as Record<string, unknown>;
     }
 
-    public resolveNamedTypeOrThrow({ typeId }: { typeId: TypeId }): DynamicSnippets.NamedType {
+    public resolveNamedType({ typeId }: { typeId: TypeId }): DynamicSnippets.NamedType | undefined {
         const namedType = this.ir.types[typeId];
         if (namedType == null) {
-            throw new Error(`Type identified by "${typeId}" could not be found`);
+            this.errors.add({
+                severity: Severity.Critical,
+                message: `Type identified by "${typeId}" could not be found`
+            });
+            return undefined;
         }
         return namedType;
     }
 
-    public resolveDiscriminatedUnionTypeInstanceOrThrow({
+    public resolveDiscriminatedUnionTypeInstance({
         discriminatedUnion,
         value
     }: {
         discriminatedUnion: DynamicSnippets.DiscriminatedUnionType;
         value: unknown;
-    }): DiscriminatedUnionTypeInstance {
-        const record = this.getRecordOrThrow(value);
+    }): DiscriminatedUnionTypeInstance | undefined {
+        const record = this.getRecord(value);
+        if (record == null) {
+            return undefined;
+        }
 
         const discriminantFieldName = discriminatedUnion.discriminant.wireValue;
         const discriminantValue = record[discriminantFieldName];
         if (discriminantValue == null) {
-            throw new Error(
-                `Missing required discriminant field "${discriminantFieldName}" got ${JSON.stringify(value)}`
-            );
+            this.errors.add({
+                severity: Severity.Critical,
+                message: this.newParameterNotRecognizedError(discriminantFieldName).message
+            });
+            return undefined;
         }
         if (typeof discriminantValue !== "string") {
-            throw new Error(`Expected discriminant value to be a string but got: ${JSON.stringify(discriminantValue)}`);
+            this.errors.add({
+                severity: Severity.Critical,
+                message: `Expected discriminant value to be a string but got: ${JSON.stringify(discriminantValue)}`
```

Comment:
`${JSON.stringify(discriminantValue)}` this might be a little tough for anyone to render in a pretty way -- maybe we just add the type of `discriminantValue` ?

Reply:
Good call, happy to adjust the message here.